### ISSUE TITLE
stb_tilemap_editor.h compilation errors

### ISSUE THREAD

**Issue by qeedquan** [nothings/stb#872]
I have the following example code using the latest stb_tilemap_editor (0.41)

```c
#include <stdio.h>
#include <stdlib.h>

#define STB_TILEMAP_EDITOR_IMPLEMENTATION

void
STBTE_DRAW_RECT(int x0, int y0, int x1, int y1, unsigned int color)
{
}

void
STBTE_DRAW_TILE(int x0, int y0, unsigned short id, int highlight, float *data)
{
}

#include "stb_tilemap_editor.h"

int
main(void)
{
	return 0;
}
```
Trying to compile this gives the errors
```
stb_tilemap_editor.h: In function ‘stbte__tile_paint’:
stb_tilemap_editor.h:2999:29: error: ‘x0’ undeclared (first use in this function)
 2999 |             STBTE_DRAW_TILE(x0,y0, (unsigned short) data[i], 0, tm->props[mapy][mapx]);
      |                             ^~
stb_tilemap_editor.h:2999:29: note: each undeclared identifier is reported only once for each function it appears in
stb_tilemap_editor.h:2999:32: error: ‘y0’ undeclared (first use in this function)
 2999 |             STBTE_DRAW_TILE(x0,y0, (unsigned short) data[i], 0, tm->props[mapy][mapx]);
      |                                ^~
stb_tilemap_editor.h: In function ‘stbte__tile_in_palette’:
stb_tilemap_editor.h:3499:30: error: ‘t’ undeclared (first use in this function)
 3499 |          STBTE_DRAW_TILE(x,y,t->id, slot == tm->cur_tile,0);

```

**Comment by nothings** [comment]
Yeah, it looks like this got badly broken more than five years, and the test I had to make sure it didn't break like this was doing the wrong thing. Looking at the history I just don't even see exactly how it went wrong, it just seems like it would never have worked.

It may be time to retire this library entirely rather than trying to figure out how to fix it, unless someone else wants to take it on.

**Comment by JoshEngebretson** [comment]
Broke Mar 4, 2019 ``` no warnings when compiling /W3``` commit  in https://github.com/nothings/stb/commit/a0b521fcf2bfbc88cf188a5d0af1a1cf47fe4d99#diff-abf9b73e358994af60923e9438afe1ee

Great library and compiles fine on previous commit and only a couple of minor commits since

Oddly removed variables declarations are cause of error:

https://github.com/nothings/stb/commit/a0b521fcf2bfbc88cf188a5d0af1a1cf47fe4d99#diff-abf9b73e358994af60923e9438afe1eeL2937

https://github.com/nothings/stb/commit/a0b521fcf2bfbc88cf188a5d0af1a1cf47fe4d99#diff-abf9b73e358994af60923e9438afe1eeL3504

**Comment by riz0ma** [comment]
I also noticed that some stuff was missing, mainly calls to stbte__hittest on all clickable buttons and stuff and some preamble x y variables on a couple of paint funtions, I got it working a month ago, but I was testing it a bit, since I'm still writing my first little 2d engine, and I needed more features for testing the library. Since I was drawing the editor ui half transparent I also noticed two texts that where drawn also in the update event, making them flicker, so I added a test in the stbte__editor_traverse function. Now It seems to work reliably, Should I "commit" my version or the diff here is enough to fix it? New to github here :)

**Comment by nothings** [comment]
I don't know the answer to that, I'll have to find some time to dig into it. Thanks both of you for helping figure out what's going on here.

**Comment by rygorous** [comment]
Took two pull request that look like they should fix this, merged into dev branch, but I still have no working example to test it with so no idea whether it actually works or not. Either way, those fixes will be in the next release, here's hoping.
